crates/manifest/src/resolved.rs

```diff
@@ -0,0 +1,219 @@
+use std::{collections::BTreeMap, path::PathBuf};
+
+use serde::{Deserialize, Serialize};
+use serde_value::Value;
+use serde_with::{hex::Hex, As};
+
+use crate::ManifestVersion;
+
+/// ResolvedManifest is a "resolved" application manifest, ready to be used by an executor.
+//
+// NOTE: The serialization of this data may be used for application de-duplication, so it
+// should be kept reasonably stable, at least for a particular Spin release.
+#[derive(Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct ResolvedManifest {
+    /// Manifest schema version.
+    pub spin_lock_version: ManifestVersion<0>,
+
+    /// Application metadata, for logging, debugging, etc.
+    pub metadata: ApplicationMetadata,
+
+    /// Custom configuration variables.
+    pub variables: BTreeMap<String, Variable>,
+
+    /// Per-trigger-type configuration. Must have an entry for every trigger type used by the application.
+    #[serde(default)]
+    pub trigger_types: BTreeMap<String, BTreeMap<String, Value>>,
```

Comment:
Will this be "slightly more typed" in the future?

Reply:
Nope, but this would get forwarded to the TriggerExecutor to deserialize into its own config model.